src/app/GitUI/Editor/Diff/DiffHighlightService.cs

```diff
@@ -148,38 +129,59 @@ private void SetText(ref string text)
         text = sb.ToString();
     }
 
+    /// <summary>
+    /// Set highlighting for <paramref name="text"/>.
+    /// The parsed added/removed lines in <see cref="_diffLinesInfo"/> is used as well as
+    /// the highlighting in <see cref="_textMarkers"/> (if Git highlighting <see cref="_useGitColoring"/>),
+    /// is used to mark inline differences (dim unchanged part of lines).
+    /// </summary>
+    /// <param name="text">The text to process.</param>
+    internal void SetHighlighting(string text)
+    {
+        if (!_useGitColoring)
+        {
+            HighlightAddedAndDeletedLines(_textMarkers);
+        }
+
+        // Apply GE word highlighting for Patch display (may apply to Difftastic setting, if not available for a repo)
+        if (!_useGitColoring || AppSettings.DiffDisplayAppearance.Value != GitCommands.Settings.DiffDisplayAppearance.GitWordDiff)
+        {
+            MarkInlineDifferences(text, _textMarkers);
+        }
```

Comment:
1. Although it is called once only, reset `_textMarkers` for clarity.
2. Although the added/deleted markes need to come first, insert them afterward in order to avoid the creation of temporary lists.
```suggestion
        _textMarkers.Clear();

        // Apply GE word highlighting for Patch display (may apply to Difftastic setting, if not available for a repo)
        if (!_useGitColoring || AppSettings.DiffDisplayAppearance.Value != GitCommands.Settings.DiffDisplayAppearance.GitWordDiff)
        {
            AddInlineDifferenceMarkers(text, _textMarkers);
        }

        if (!_useGitColoring)
        {
            InsertAddedAndDeletedLineMarkers(_textMarkers);
        }

```

Reply:
With Git coloring (default), the text markers are set when parsing the document so insert is normally required. Only part of the suggestions can be used.
But changing to insert without adding to temp list.

Reply:
> With Git coloring (default), the text markers are set when parsing the document so insert is normally required.

My suggestion still is, to add the in-line markers in forward order _first_.
Then the added/deleted markers are _inserted_ at position 0 into `_textMarkers`. (This reverses the lines. But this should not matter.)
This is mainly achieved by reversing the execution order in `SetHighlighting`, which you have already applied.

Having `insertPos = 0` all the time for in-line markers does not have any effect. Furthermore, the `insertPos` has not been applied to all in-line markers.
I think `insertPos` should be removed from the in-line diff functions - but be applied to the added/deleted lines functions.

Reply:
The text markers for the normal marking is added first, in SetText. Before that, the text has the escape sequences.
I changed so SetHighlighting() is where insertion is done, with the comment there instead